draft-irtf-cfrg-bbs-signatures.md

```diff
@@ -786,78 +1046,54 @@ ABORT if:
+Each Interface MUST also define a unique ID as a parameter, called `api_id`. It is REQUIRED from the operations that create generators and map messages to scalars, to also define a unique ID (see (#interface-utilities)). The `api_id` MUST have the following format:
 
 ```
-random_scalars = calculate_random_scalars(count)
-
-Inputs:
-
-- count (REQUIRED), non negative integer. The number of pseudo random
-                    scalars to return.
-
-Parameters:
-
-- get_random, a pseudo random function with extendable output, returning
-              uniformly distributed pseudo random bytes.
-- expand_len, defined by the ciphersuite.
-
-Outputs:
-
-- random_scalars, a list of pseudo random scalars,
-
-Procedure:
-
-1. for i in (1, ..., count):
-2.     r_i = OS2IP(get_random(expand_len)) mod r
-3. return (r_1, r_2, ..., r_count)
+api_id = ciphersuite_id || CREATE_GENERATORS_ID || MAP_TO_SCALAR_ID || ADD_INFO
 ```
 
-## Generators Calculation
+Where `ciphersuite_id` is defined by the ciphersuite, `CREATE_GENERATORS_ID` is the unique IDs of the operation that creates the generators, `MAP_TO_SCALAR_ID` is the unique ID of the operation that maps the messages to scalars and the `ADD_INFO` value is an optional octet string indicating any additional information used to uniquely qualify the Interface. When `ADD_INFO` is present, it MUST only contain ASCII encoded characters with codes between 0x21 and 0x7e (inclusive) and MUST end with an underscore (ASCII code: 0x5f), other than the last character the string MUST not contain any other underscores (ASCII code: 0x5f). The `api_id` value, MUST be used by all subroutines an Interface calls, to ensure proper domain separation.
 
-A `create_generators` procedure defines how to create a set of randomly sampled points from the G1 subgroup, called the generators. Generators form a part of the public parameters used by the BBS Signature scheme to accomplish operations such as [Sign](#signature-generation-sign), [Verify](#signature-verification-verify), [ProofGen](#proof-generation-proofgen) and [ProofVerify](#proof-verification-proofverify). A `create_generators` operation takes as input the following arguments,
+Interfaces are meant to make it easier to use BBS Signature as part of other protocols with different requirements (for example, different types of input messages), or to extend BBS Signatures with additional functionality. Documents defining new BBS Interfaces, are REQUIRED to include a detailed and peer reviewed analyses, showcasing that, under reasonable cryptographic assumptions, the documented scheme is secure under the required security definitions and threat model of each protocol. In other words, Interfaces MUST be treated like Ciphersuites ((#ciphersuites)), in the sense that it is RECOMMENDED that applications will avoid creating their own, proprietary Interfaces.
```

Comment:
Feels like there should be more elaboration here, regarding safe extension.

Also not sure if there is a reference we can use to help send this signal home, perhaps ask the list for guidance on this.

Reply:
Agreed. Made the sentence informational, expanded a bit and added some references. Will also ask the list for guidance.